src/lib/vocab/community/communityHelpers.server.ts

```
@@ -63,118 +59,79 @@ export const initAdminCommunity = async (
-			// Create the default role and assign it
-			role = unwrap(
-				// eslint-disable-next-line no-await-in-loop
-				await initDefaultRoleForCommunity(repos, community, roleTemplate.name),
-			);
 			defaultRoleId = role.role_id;
-		} else {
-			role = unwrap(
-				// eslint-disable-next-line no-await-in-loop
-				await repos.role.create(community.community_id, roleTemplate.name),
-			);
-			if (roleTemplate.creator) {
-				creatorRoleId = role.role_id;
-			}
+			community.settings.defaultRoleId = defaultRoleId;
+			unwrap(await repos.community.updateSettings(community.community_id, community.settings)); // eslint-disable-line no-await-in-loop
+		}
+		if (roleTemplate.creator) {
+			creatorRoleId = role.role_id;
 		}
-		roles.push(role);
 
 		if (roleTemplate.policies) {
+			// TODO batch with a repo method
 			for (const policyTemplate of roleTemplate.policies) {
-				const policy = unwrap(
-					// eslint-disable-next-line no-await-in-loop
-					await repos.policy.create(role.role_id, policyTemplate.permission),
-				);
-				policies.push(policy);
+				policies.push(unwrap(await repos.policy.create(role.role_id, policyTemplate.permission))); // eslint-disable-line no-await-in-loop
 			}
 		}
 	}
-	let creatorAssignment: Assignment;
-	if (creatorRoleId) {
-		creatorAssignment = unwrap(
-			await repos.assignment.create(actor, community.community_id, creatorRoleId),
-		);
-	} else if (defaultRoleId) {
-		creatorAssignment = unwrap(
-			await repos.assignment.create(actor, community.community_id, defaultRoleId),
-		);
-	} else {
-		creatorAssignment = unwrap(
-			await repos.assignment.create(actor, community.community_id, roles[0].role_id),
-		);
-	}
-	return {ok: true, value: {roles, creatorAssignment, policies}};
+
+	// TODO in some cases we probably want to add both the `creatorRoleId` and `defaultRoleId` assignment, how to express that in the template data?
+	const creatorAssignmentRoleId = creatorRoleId || defaultRoleId || roles[0].role_id;
```

several Qs here:

- sometimes we may want to give a default role to the creator, but other times not (e.g. "member" and "newcomer") - I'm not sure how we could express this -- that's why I changed the API to an array, which I think is the right call, but I balked at changing existing behavior and left the todo
- isn't having a `defaultRoleId` an invariant we expect elsewhere, so we should expect it here too? As written, this helper now gets called for all community types, so it can be safely used for validation
- on the subject of validation, it's a little strange that this silently ignores cases like multiple `creator`/`default` templates, creating only exactly one assignment

---

Maybe instead of designating creator/default at the Role level, like you say put it at a higher level of the template interface would probably be the most useful, especially if that gives us some type of guarantee of a `defaultRoleId`. Maybe couple that with `creatorRole` actually being `creatorRoles` with an array of roles to assign and if it's empty just use the defaultRole?

---

yeah that sounds good, and this PR is so big that I'll get it merged and focus on the followups next